baseplate/integration/pyramid.py

```diff
@@ -86,18 +86,38 @@ class ServerSpanInitialized(BaseplateEvent):
     pass
 
 
+class HeaderTrustHandlerInterface(object):
+    """Class used by BaseplateConfigurator to validate headers.
+    """
+
+    def should_trust_trace_headers(self, request):
+        raise NotImplementedError
+
+
+class StaticTrustHandler(HeaderTrustHandlerInterface):
+    def __init__(self, trust_headers=False):
+        self.trust_headers = trust_headers
+
+    def should_trust_trace_headers(self, request):
+        return self.trust_headers
+
+
 # pylint: disable=abstract-class-not-used
 class BaseplateConfigurator(object):
     """Config extension to integrate Baseplate into Pyramid.
 
     :param baseplate.core.Baseplate baseplate: The Baseplate instance for your
         application.
-    :param bool trust_trace_headers: Should this application trust trace
+    :param bool trust_trace_headers: (DEPRECATED) Should this application trust trace
         headers from the client? If ``True``, trace headers in inbound requests
         will be used for the server span. If ``False``, new random trace IDs
         will be generated for each request.
     :param baseplate.core.EdgeRequestContextFactory edge_context_factory: A
         configured factory for handling edge request context.
+    :param baseplate.integration.pyramid.HeaderTrustHandlerInterface header_trust_handler:
+        An object which will be used to verify whether the headers for a request
```

Comment:
this sorta sounds like any headers would be involved in this. how about:

"to verify whether baseplate should parse request context headers like trace IDs"